languages/cpp/examples/Wrapper.cpp

```diff
@@ -18,57 +22,110 @@ int main() {
     std::string accessToken = accessTokenEnv;
     std::string organizationId = organizationIdEnv;
 
-    // Configuring the URLS is optional, remove them to use the default values
+    // Configuring the URLS is optional; if unset, use bitwarden.com
     BitwardenSettings bitwardenSettings;
-    bitwardenSettings.set_api_url(apiUrl);
-    bitwardenSettings.set_identity_url(identityUrl);
+    if (apiUrl != nullptr && identityUrl != nullptr) {
+        bitwardenSettings.set_api_url(apiUrl);
+        bitwardenSettings.set_identity_url(identityUrl);
+    } else {
+        std::cerr << "Info: API_URL and IDENTITY_URL are not set, using default values..." << std::endl;
+    }
 
     // Create a Bitwarden client instance
-    BitwardenClient bitwardenClient = BitwardenClient(bitwardenSettings);
-    // // Access token login
-    bitwardenClient.loginAccessToken(accessToken);
-    // Organization ID
-    boost::uuids::uuid organizationUuid = boost::uuids::string_generator()(organizationId);    
-    
-    // // Create a new project
+    BitwardenClient bitwardenClient(bitwardenSettings);
+
+    // Access token login
+    if (stateFile != nullptr) {
+        bitwardenClient.loginAccessToken(accessToken, stateFile);
+    } else {
+        bitwardenClient.loginAccessToken(accessToken);
```

Comment:
Question: as far as I know we only have one `loginAccessToken` function defined, that takes `(const std::string& accessToken, const std::string& stateFile)`, so what does just passing an access token do?

Reply:
Good question! If you were to do something like this:

```c++
const char *stateFile = std::getenv("SOME_VAR_THAT_DOESNT_EXIST");
// since the var doesnt exist, getenv sets stateFile to a nullptr
bitwardenClient.loginAccessToken(accessToken, stateFile);
```

You'll end up with a segmentation fault. Perhaps we could handle that in the `loginAccessToken` function? I'm not sure what the idiomatic thing to do is. I imagine C++ devs might be used to having to check for `nullptr`s with `getenv`, but I'm not sure honestly.

Reply:
Makes sense, yeah I guess this is one of the many ugly warts from C++ we're dealing with. I think leaving it as-is is fine